File: DSA Unit 05/Lec 33/Q1.cpp
Single File Programming Question
Problem Statement



Ben,a software developer, is working on a hash table implementation using linear probing for collision resolution. He needs to insert several keys into the hash table, display the final state of the hash table, and check if the 3rd position is empty or contains a value. If the position is empty, the program should print that it's empty; otherwise, it should print the value stored at that position.



Help him to implement the task.



FORMULA:

Primary hash function: hash(key) = key % TABLE_SIZE

Linear probing formula: index = (index + 1) % TABLE_SIZE

Input format :
The first line of input consists of integer numKeys, representing the number of keys to be inserted into the hash table.

The next numKeys lines each contain an integer key, representing the keys to be inserted into the hash table.

Output format :
The first line of output prints: Hash Table: <state_of_the_table>"

"The second line of output prints: Position 3 is Empty" or "Value at position 3: <value>



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

≤ numKeys ≤ 6

-10^6 ≤ key ≤ 10^6

The size of the hash table is fixed at 6.

Sample test cases :
Input 1 :
5
12 18 24 30 36
Output 1 :
Hash Table: [12] [18] [24] [30] [36] [ ] 
Value at position 3: 30
Input 2 :
4
5 10 15 20
Output 2 :
Hash Table: [ ] [ ] [20] [15] [10] [5] 
Value at position 3: 15
Input 3 :
4
10 20 30 40
Output 3 :
Hash Table: [30] [ ] [20] [ ] [10] [40] 
Position 3 is Empty


#include <bits/stdc++.h>
using namespace std;

int main() {
    int numKeys;
    cin >> numKeys;

    const int TABLE_SIZE = 6;
    vector<long long> table(TABLE_SIZE, LLONG_MIN);

    for (int i = 0; i < numKeys; i++) {
        long long key;
        cin >> key;

        int index = ((key % TABLE_SIZE) + TABLE_SIZE) % TABLE_SIZE;

        while (table[index] != LLONG_MIN) {
            index = (index + 1) % TABLE_SIZE;
        }
        table[index] = key;
    }

    cout << "Hash Table: ";
    for (int i = 0; i < TABLE_SIZE; i++) {
        if (table[i] == LLONG_MIN)
            cout << "[ ] ";
        else
            cout << "[" << table[i] << "] ";
    }
    cout << "\n";

    if (table[3] == LLONG_MIN)
        cout << "Position 3 is Empty";
    else
        cout << "Value at position 3: " << table[3];

    return 0;
}

File: Java/Unit 02/Static/Q1.java
Single File Programming Question
Problem Statement:



Elisa, a language processing specialist, is working on a project that requires 
  converting strings to character arrays. To assist her in this task, you are tasked 
  with creating a program that converts a given string to a character array.This program will 
  help Elisa efficiently manipulate and analyze text data by converting it into a more versatile format.

Input format :
The input contains a string representing the string to be converted as a character array.

Output format :
The output prints the characters of the given string in the form of a character array.

Each character is separated by a comma and a space.
The array is enclosed within square brackets [ ].
Spaces in the input string should be preserved and displayed as empty spaces between commas.


Refer to the sample output for format specifications

Code constraints :
The input string should not be empty.

The input string may contain alphanumeric characters (uppercase and lowercase), spaces and special characters.

The input string's length can be up to 105 characters.

Sample test cases :
Input 1 :
Hello
Output 1 :
[H, e, l, l, o]
Input 2 :
The quick brown fox jumps over the lazy dog.
Output 2 :
[T, h, e,  , q, u, i, c, k,  , b, r, o, w, n,  , f, o, x,  , j, u, m, p, s,  , o, v, e, r,  , t, h, e,  , l, a, z, y,  , d, o, g, .]
Input 3 :
123ABC!@#
Output 3 :
[1, 2, 3, A, B, C, !, @, #]





  import java.util.Scanner;
import java.util.Arrays;

public class Main {
    // You are using Java
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        // Read full input line
        String input = sc.nextLine();

        // Convert string to character array
        char[] characters = input.toCharArray();

        // Print in required format
        System.out.print("[");

        for (int i = 0; i < characters.length; i++) {
            System.out.print(characters[i]);

            if (i != characters.length - 1) {
                System.out.print(", ");
            }
        }

        System.out.print("]");

        sc.close();
    }
}
